tests/testscript.py

```diff
@@ -1447,6 +1624,35 @@ def number_of_storage_devices(machine):
     return int(out)
 
 
+def pci_devices_by_bdf(machine):
+    """
+    Creates a dict of all PCI devices addressable by their BDF in the VM.
+
+    BDFs are keys, while the combination of vendor and device IDs form the
+    associated value.
+
+    :param machine: Host machine of the nested VM
+    :return: BDF mapped to devices, example: {'00:00.0': '8086:0d57'}
+    :rtype: dict[str, str]
+    """
+    status, lines = ssh(
+        machine,
+        "lspci -n | awk '/^[0-9a-f]{2}:[0-9a-f]{2}\\.[0-9]/{bdf=$1}{class=$3} {print bdf \",\" class}'",
+    )
+    assert status == 0
+    out = {}
+    for line in lines.splitlines():
+        bdf, device_class = line.split(",")
+        out[bdf] = device_class
+    return out
+
+
+def reset_system_image(machine):
```

Comment:
```suggestion
def reset_system_image(machine):
    """
    Replaces the (possibly modified) system image with its original image.
    
    This helps avoid situations where a VM hangs during boot after the underlying disk’s BDF was changed, since OVMF may store NVRAM entries that reference specific BDF values.
    """
```